fossilize_external_replayer_linux.hpp

```diff
@@ -49,10 +49,59 @@
 #include "platform/gcc_clang_spinlock.hpp"
 #endif
 
+#ifdef __linux__
+
 #ifndef SCHED_BATCH
 #define SCHED_BATCH 3
 #endif
 
+/**
+ * Define the syscall interface in Linux because it is missing from glibc
+ * https://www.kernel.org/doc/html/latest/block/ioprio.html
+ */
+
+#ifndef IOPRIO_CLASS_SHIFT
+#define IOPRIO_CLASS_SHIFT (13)
+#endif
+
+#ifndef IOPRIO_PRIO_MASK
+#define IOPRIO_PRIO_MASK ((1UL << IOPRIO_CLASS_SHIFT) - 1)
+#endif
+
+#ifndef IOPRIO_PRIO_CLASS
+#define IOPRIO_PRIO_CLASS(mask) ((mask) >> IOPRIO_CLASS_SHIFT)
+#endif
+
+#ifndef IOPRIO_PRIO_DATA
+#define IOPRIO_PRIO_DATA(mask) ((mask) & IOPRIO_PRIO_MASK)
```

Comment:
Ditto

Reply:
Yeah, those are for the get function to split the value back into class and prio.